src/app/api/notification/notification.service.ts

```diff
@@ -0,0 +1,104 @@
+import { CopilotAPI } from '@/utils/CopilotAPI'
+import { BaseService } from '@api/core/services/base.service'
+import { AssigneeType, Task } from '@prisma/client'
+import { NotificationTaskActions } from '../core/types/tasks'
+import { z } from 'zod'
+import { CompanyResponse, CopilotUser, MeResponse } from '@/types/common'
+import { getEmailDetails, getInProductNotificationDetails } from './notification.helpers'
+
+export class NotificationService extends BaseService {
+  async create(action: NotificationTaskActions, task: Task) {
+    try {
+      const copilotUtils = new CopilotAPI(this.user.token)
+
+      const { senderId, recipientId, actionUser } = await this.getNotificationParties(copilotUtils, task, action)
+
+      const notificationDetails = {
+        senderId,
+        recipientId,
+        deliveryTargets: {
+          inProduct: getInProductNotificationDetails(actionUser)[action],
+          email: getEmailDetails(actionUser)[action],
+        },
+      }
+
+      await copilotUtils.createNotification(notificationDetails)
+    } catch (error) {
+      console.error(`Failed to send notification for action: ${action}`, error)
+    }
+  }
+
+  async createBulkNotification(action: NotificationTaskActions, task: Task, recipientIds: string[]) {
+    try {
+      const copilotUtils = new CopilotAPI(this.user.token)
+      const userInfo = await copilotUtils.me()
+      const senderId = z.string().parse(userInfo?.id)
+      const actionUserName = `${userInfo?.givenName} ${userInfo?.familyName}`
```

Comment:
@aatbip rather that optional chaining to `userInfo` it would be better if we throw when `userInfo` is `null`, that would make code more clear and concise IMO, let me know your thoughts

Reply:
Fixed in 3a9dcb86db254fe7b133359ecb68d8b0b1d765be commit